Slim/Control/Queries.pm

```diff
@@ -4734,25 +4740,40 @@ sub worksQuery {
 		$sql .= ' ';
 	}
 
-	$sql .= " GROUP BY $groupBy ";
-
 	my $dbh = Slim::Schema->dbh;
 
 	# Get count of unique composers, the count is cached until the next rescan done event
-	my $cacheKey = md5_hex($sql . join( '', @{$p} ) . 'composerCount' . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
+	my $ccSql = sprintf($sql . " GROUP BY composer.id", "'c'");
+	my $cacheKey = md5_hex($ccSql . join( '', @{$p} ) . Slim::Music::VirtualLibraries->getLibraryIdForClient($client));
```

Comment:
Hmm... I thought I had commented on this already? Was it in a different PR?

You define `my $cacheKey` twice within the same scope, but never actually populate it. Therefore the check for cached data will always fail.

Reply:
Surely it's being populated at 4747 (and 4764)?

Agreed it doesn't need `my ` at 4764.

Reply:
The variable `$cacheKey` is defined, yes. But there's no cache where it's being used. You only read from the cache, but never write to `$cache{$cacheKey}`, do you?

Reply:
Ah, I see. I copied yearsquery as a basis for worksquery - it's not there either! Looking at other queries, I can see what needs to be done.

Reply:
Oh my... thanks for the heads up!

Reply:
Changes pushed.